### PR TITLE
Add post processor to register ProviderSettings @Bean

### PR DESCRIPTION
**Pull request by sjohnr**

### CODE REVIEW

File: oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+	private BeanFactory beanFactory;
+
+	@Override
+	public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) throws BeansException {
+		for (AbstractBeanDefinition beanDefinition : beanDefinitions) {
+			String[] beanNames = BeanFactoryUtils.beanNamesForTypeIncludingAncestors(
+					(ListableBeanFactory) this.beanFactory, beanDefinition.getBeanClass(), false, false);
+			if (beanNames.length == 0) {
+				String beanName = beanNameGenerator.generateBeanName(beanDefinition, registry);
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java:54]: I wonder if we need a `beanNameGenerator`? Can we simply use `beanDefinition.getBeanClassName()` for  the bean name?

File: oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+final class AuthorizationServerBeanDefinitionRegistryPostProcessor implements BeanDefinitionRegistryPostProcessor, BeanFactoryAware {
+
+	private final AnnotationBeanNameGenerator beanNameGenerator = new AnnotationBeanNameGenerator();
+	private final List<AbstractBeanDefinition> beanDefinitions = new ArrayList<>();
+
+	private BeanFactory beanFactory;
+
+	@Override
+	public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) throws BeansException {
+		for (AbstractBeanDefinition beanDefinition : beanDefinitions) {
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java:50]: Prefix `this` -> `this.beanDefinitions`

File: oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.context.annotation.AnnotationBeanNameGenerator;
+
+/**
+ * Post processor to register one or more bean definitions on container initialization, if not already present.
+ *
+ * @author Steve Riesenberg
+ * @since 0.2.0
+ */
+final class AuthorizationServerBeanDefinitionRegistryPostProcessor implements BeanDefinitionRegistryPostProcessor, BeanFactoryAware {
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java:41]: Let's rename to `RegisterMissingBeanPostProcessor`

File: oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessor.java:2]: Copyright should be `2020-2021`

File: oauth2-authorization-server/src/test/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessorTests.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
```
Comment (jgrandja) [inline oauth2-authorization-server/src/test/java/org/springframework/security/config/annotation/web/configuration/AuthorizationServerBeanDefinitionRegistryPostProcessorTests.java:2]: Copyright should be `2020-2021`

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @sjohnr ! Please see review comments.

**Comment by sjohnr** [comment]
@jgrandja Updates ready for review.

**Comment by jgrandja** [comment]
Thanks for the PR @sjohnr ! This is now in main.

**Pull request closed without merging**